### ISSUE TITLE
Name with special characters filtering doesn't work In GraphQl Product API

### ISSUE THREAD

**Issue by ShafeelSha** [magento/magento2#31464]
### Preconditions (*)

1. Magento 2.3.5-p2
2. Create a product that **name should include at least a special character**(ex: **Alkaline Phosphatase (SL) - IFCC** )

### Steps to reproduce (*)

1. A product that name should include at least a special character (ex: Alkaline Phosphatase (SL) - IFCC )
2. Open Altair or any graphql client , then try products filter query _(below in my case)_
```
{
  products(filter: { name: { match: "Alkaline Phosphatase (SL) - IFCC%" } }) {
    items {
      name
      id
      sku
    }
  }
}
```

### Expected result (*)

1.  Expected Output
![image](https://user-images.githubusercontent.com/53746963/103149243-b8200880-478d-11eb-803c-ef5738ff505c.png)

### Actual result (*)

1. Current output
![image](https://user-images.githubusercontent.com/53746963/103149258-df76d580-478d-11eb-9ef9-6e4c1fff9388.png)

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by Pavel-Fekete** [comment]
Not reproducible on the latest 2.4-develop
![grapql](https://user-images.githubusercontent.com/67371013/103203366-67c7b880-48fd-11eb-9bc4-4e63ebc9724e.png)

**Comment by engcom-Bravo** [comment]
Hello @ShafeelSha 

Thank you for your report

We are not able to reproduce the issue on the latest 2.4-develop.

We have created a product named _"Alkaline Phosphatase (SL) - IFCC"_
![prod_name](https://user-images.githubusercontent.com/51680745/103207519-37851780-4907-11eb-857b-aa0d70d8133e.png)

And when we perform a GraphQL query
```graphql
{
  products(filter: { name: { match: "Alkaline Phosphatase (SL) - IFCC%" } }) {
    items {
      name
      id
      sku
    }
  }
```
the response is
```graphql
{
  "data": {
    "products": {
      "items": [
        {
          "name": "Alkaline Phosphatase (SL) - IFCC",
          "id": 2,
          "sku": "Alkaline Phosphatase (SL) - IFCC"
        }
      ]
    }
  }
}
```
![graph_query](https://user-images.githubusercontent.com/51680745/103207593-669b8900-4907-11eb-86ea-cb887ed47f1c.png)

So we are closing this issue. Please feel free to comment, reopen or create a new ticket according to the [Issue reporting guidelines](https://github.com/magento/magento2/wiki/Issue-reporting-guidelines)

**Comment by ShafeelSha** [comment]
@engcom-Bravo @Pavel-Fekete  Thanks for the replay , the issue was Magento 2.3.5-p2 , i think that was  fixed on Magento 2.4, let me try with latest version
